practices/follow-functional-core-imperative-shell.md

```diff
@@ -3,117 +3,153 @@
+
+The Functional Core, Imperative Shell practice and object-oriented programming (OOP) principles are not mutually exclusive.
+As long as the functional core remains free of side effects, it can co-exist within an object-oriented architecture.
+The functional core's logic could be encapsulated within methods of a class, ensuring that these methods return new objects instead of mutating their fields.
+The functional core can take advantage of polymorphism by receiving or returning objects of a class hierarchy with multiple subclasses to represent various inputs or outputs. <ed> I'm assuming your target reader will follow along with this explanation just fine. But this is one example of where this practice reads (to me) much more technical than others. What might help is relating this nuance back to code maintainability. </ed> 
+Developers can implement the functional core while adhering to OOP principles, taking advantage of both paradigms.
+
+### Imperative Shell Will Perform Various Tasks
 
-### There can not be any logic in the Imperative Shell
+<ed> I'm not sure I've accurately captured the nuance/pitfall in the subhead title above. WDYT? </ed> 
 
 In simple cases, the imperative shell merely passes inputs to the functional core, receives the response, and renders it back to the user.
 However, there are scenarios where the functional core may produce outputs that require inspection or processing by the imperative shell.
-While the imperative shell should ideally remain minimal and devoid of complex logic, it may need to analyze, interpret, or transform the response from the functional core to ensure proper interaction and presentation to the user.
+While the imperative shell should ideally remain simple and devoid of complex logic, it may need to analyze, interpret, or transform the response from the functional core to ensure proper interaction and presentation to the user.
 
 ### Side Effects should only happen before and after the execution of the Functional Core
 
-Ideally, the imperative shell should organize the IO (Input/Output) parts of the code at the beginning and end of its execution, sandwiching the invocation of the Functional Core in the middle, following the pattern IO => FC => IO.
-However, in certain scenarios, there may be a need to interleave IO operations with calls to the Functional Core, resulting in a pattern such as IO => FC => IO => FC => IO.
-While this deviation from the ideal pattern may occur out of necessity, it should be minimized whenever possible.
+<ed> For consistency with other subheads under Nuances, I'd love to simplify and shorten this title. We don't mention the term "side effects" below, though. Are "side effects" synonymous with "I/O operations"? Perhaps "I/O Operations Sometimes Call Functional Core"? </ed> 
```

Comment:
Let's instead just remove this. It's a detail. Not worthy of the real estate it's currently taking on the page.

Reply:
Ok, just so I understand, we're removing the subhead "### Side Effects should only happen before and after the execution of the Functional Core" and all text that's under it?

Reply:
Correct.

Reply:
Done.